extensions/api/data-management/contractnegotiation/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractnegotiation/model/NegotiationInitiateRequestDto.java

```diff
@@ -46,6 +49,14 @@ public ContractOfferDescription getOffer() {
         return offer;
     }
 
+    @AssertTrue(message = "connectorId, connectorAddress, protocol, offerId and assetId cannot be blank")
+    @JsonIgnore
+    public boolean isValid() {
```

Comment:
Why do we need this? all properties are already validated using `@NotNull` annotations, right? Even the nested ones?

Reply:
I moved the `isValid` method from here:
https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/501883cc32f7d390b1c3bbad562c551210dbfa88/extensions/api/data-management/contractnegotiation/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractnegotiation/ContractNegotiationApiController.java#L175

plus, there's something wrong with the jersey-bean-validation, the validation is not working every time, before this PR we had all the NPE becoming 400, but changing this behavior made me notice that actually there are a lot of NPE thrown because of non-working validation, I need to investigate more